### ISSUE TITLE
API-22: MP FaultTolerance 2.0 Epic

### ISSUE THREAD

**Issue by Emily-Jiang** [OpenLiberty/open-liberty#4080]
Implements MicroProfile FT 1.2 release

Support the return type of CompletionStage when annotated with Asynchronous
https://github.com/eclipse/microprofile-fault-tolerance/issues/291
https://github.com/eclipse/microprofile-fault-tolerance/issues/110

**Comment by Emily-Jiang** [comment]
https://bigblue.aha.io/features/API-22

**Comment by marikaj123** [comment]
not in 19200

**Comment by marikaj123** [comment]
Does the feature has English messages that require translation?

**Comment by Azquelt** [comment]
It does include English messages. Looking at the files, it looks like they've already been translated.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by marikaj123** [comment]
@Emily-Jiang - Please provide a link to the AHA feature

**Comment by Azquelt** [comment]
@marikaj123 I didn't realise we were supposed to put that in github. Here it is: https://bigblue.aha.io/features/API-22

**Comment by gscottj** [comment]
Accessibility Approval Comment:  
Based on the description provided, it appears that this feature does not have any end-user user interfaces that require accessibility compliance testing. End user user-interfaces include GUI, CLI and documentation that users are expected to interact with to accomplish tasks or goals. If this assumption is not correct, then let me know that required accessibility verification testing of the UI will be performed before it is shipped and what are the results from those tests.

**Comment by Azquelt** [comment]
@skasund STE slides available here: https://ibm.box.com/s/wwluwush1t1knmsrhicf7fxz8u13o74j

**Comment by skasund** [comment]
Thanks for the slides.  I've provided the STE approval.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by atosak** [comment]
@donbourne , Can we have serviceability approval for this feature. See the response above for the serviceability questions.

**Comment by atosak** [comment]
ID:
https://github.com/OpenLiberty/docs/issues/136
https://github.com/OpenLiberty/open-liberty/issues/4078

**Comment by donbourne** [comment]
@Emily-Jiang can you please ask SVT team if they are ok with serviceability -- (question 3b)?

**Comment by atosak** [comment]
@hanczaryk , Can you provide an answer to Don's question above (Question 3b), Thank you.

**Comment by hanczaryk** [comment]
Sorry for any delay, but here are the SVT Serviceability Approval Comments.
a) Who conducted SVT tests for this feature? Brian Hanczaryk
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  Yes, I agree that serviceability was sufficient.

**Comment by sabolo** [comment]
I have been informed that the automated testing for this feature has been disabled due to reversion of a fix to the arquilian container.  I am pulling the approval for FAT as a result as currently the automated testing is not enabled and as such delivered with the feature.

**Comment by sabolo** [comment]
I will grant approval once the testing is re-enabled

**Comment by sabolo** [comment]
I have had a chat with @tevans78 about the issue.  The impact is that 2 of the TCK tests are not automated until the fix and although its been run manually we have not verified cross platform.   Given the manual testing and the fact this has previously been working I am happy that the risk for the release is minimal and would recommend putting on the train anyway.  The FAT approval however will not be granted until the testing is fixed to ensure the technical debt is completed.

**Comment by Azquelt** [comment]
I've merged the arquillian container update and enabled the two TCK tests under #7063 which was merged yesterday.

Hasn't made it into master yet but I can see the tests passing in the continuous build.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi @Azquelt, thanks for your update, sorry I didn't spot it sooner; also thanks for taking a moment to discuss defect 261441 which was revealed in your fixing of the Arquillian Container. Given that the TCKs are now running and that said defect is a test bug I'm adding the FAT Focal Approval at this time.
